zkvm/src/prover.rs

```diff
@@ -1,25 +1,44 @@
 use bulletproofs::r1cs;
+use bulletproofs::{BulletproofGens, PedersenGens};
 use curve25519_dalek::ristretto::CompressedRistretto;
 use curve25519_dalek::scalar::Scalar;
+use merlin::Transcript;
+use std::collections::VecDeque;
 
 use crate::errors::VMError;
 use crate::ops::Instruction;
 use crate::point_ops::PointOp;
+use crate::signature::Signature;
 use crate::types::*;
-use crate::vm::{Delegate, RunTrait};
+use crate::vm::{Delegate, RunTrait, Tx, VM};
 
 pub struct Prover<'a, 'b> {
     signtx_keys: Vec<Scalar>,
     cs: r1cs::Prover<'a, 'b>,
 }
 
-pub struct RunProver {}
+pub struct RunProver {
+    program: VecDeque<Instruction>,
+}
 
 impl<'a, 'b> Delegate<r1cs::Prover<'a, 'b>> for Prover<'a, 'b> {
     type RunType = RunProver;
 
-    fn commit_variable(&mut self, com: &Commitment) -> (CompressedRistretto, r1cs::Variable) {
-        unimplemented!()
+    fn commit_variable(
+        &mut self,
+        com: &Commitment,
+    ) -> Result<(CompressedRistretto, r1cs::Variable), VMError> {
+        let (v, v_blinding) = match com {
+            Commitment::Open(w) => {
+                let val = match w.value {
```

Comment:
Can use `w.value.into()` here, since ScalarWitness already knows how to convert to Scalar.